gnat2goto/driver/tree_walk.adb

```diff
@@ -4095,6 +4113,61 @@ package body Tree_Walk is
       return  Ret;
    end Do_Signed_Integer_Definition;
 
+   ----------------------------------
+   -- Do_Floating_Point_Definition --
+   ----------------------------------
+
+   function Do_Floating_Point_Definition (N : Node_Id) return Irep is
+      E : constant Entity_Id := Defining_Entity (Parent (N));
+
+      --  This determines if ranges were specified or not
+      Scalar_Range_Ent : constant Node_Id := Scalar_Range (E);
+      Ret : constant Irep :=
+        New_Irep (if Nkind (Scalar_Range_Ent) = N_Real_Range_Specification
+                  then I_Bounded_Floatbv_Type
+                  else I_Floatbv_Type);
+      Esize_Width : constant Nat := UI_To_Int (Esize (E));
+      Digits_Expr : constant Node_Id := Digits_Expression (N);
+      Digits_Count : constant Uint := Intval (Digits_Expr);
+   begin
+      --  The digit precision is not used yet: this test is here only to show
+      --  how to access it.
+      if Digits_Count > 100 then
```

Comment:
I'd remove this before merging, doesn't seem to actually have a function. For what it's worth I don't think we need to care too much about the precision `digits`; The GNAT frontend will have picked a representation for us, and so for us it's just a matter of mapping that representation to our float types.

Reply:
Done.